src/routes/login/+page.svelte

```diff
@@ -21,12 +25,25 @@
     };
 
   $: if (form?.success) currentUser.set(form?.currentUser);
+
+  let redirectPromise: Promise<void> | undefined;
+  $: redirectURL = $page.url.searchParams.get("state");
+  $: if (form?.success && form?.currentUser) currentUser.set(form.currentUser);
+  $: if (browser && form?.success && redirectURL && !redirectPromise) {
+    redirectPromise = (async () => {
+      await timeout(500);
+      await goto(redirectURL, { replaceState: true, invalidateAll: true });
+    })();
+  }
 </script>
 
 <div class="grid-container">
   {#if form?.success}
     {#if form.currentUser}
       <p>Successfully logged in!</p>
+      {#if redirectURL}
+        <p>Please wait to be redirected to the previous page.</p>
+      {/if}
```

Comment:
Nit: In an ideal situation, this would hopefully be visible for a very short amount of time.  A UX improvement could be to have this appear in an accessible modal, along with providing a `redirectURL` link in case the redirect doesn't happen for some reason. I'd leave that for follow-up work for now.

Reply:
This is already visible for a very short amount of time - the `await timeout(500)` above does that. I've added a manual redirect link in case that doesn't work.